### DISCUSSION TITLE
TS-based concealing

### DISCUSSION THREAD

**Discussion by mrossinek** [nvim-treesitter/nvim-treesitter#1825] [Ideas]
This is mostly a question to query the status/plans for implementing Treesitter-powered concealing of characters (e.g. basic plain text markup). I am not sure how exactly the treesitter integration with Neovim works hence this is a discussion. Is something like this planned? Over at [Neorg](https://github.com/nvim-neorg/neorg) this would enable us to do some really cool and powerful stuff and fully ditch Vim's regex-based syntax definitions.

**Comment by stsewd** [comment]
I think this mostly depends on neovim implementing conceal on extmarks

**Reply by mrossinek** [reply]
Okay that makes sense. Are you aware of any ongoing work on this by any chance? Or would it be appropriate to start a discussion/issue over there directly?

**Reply by theHamsta** [reply]
I've experimented a bit with `overlay` virtual text. It will not be able to reduce the number of characters but you can hide draw of stuff like `*` to hide them. Real conceal would be much cooler.

**Reply by stsewd** [reply]
@mrossinek sorry I'm not aware if there is an open issue or PR related to this, but it's worth it asking on neovim

**Reply by mrossinek** [reply]
Thanks! For reference: https://github.com/neovim/neovim/issues/15706

@theHamsta Thanks for the input! Am I understanding this right that for the time being you would "conceal" the markup by overwriting the actual char with something different? E.g. `*text*` could be overwritten by `\ text\ ` (the backslashes are required for markdown to not remove the whitespace which I want to show here) whereby the markup itself is hidden but you would get additional whitespace instead?

**Reply by theHamsta** [reply]
yes, you can effectively display a different text than the one that's actually there. The only restriction is at the moment that you have to use the same number of characters.

**Reply by mrossinek** [reply]
Okay I see. I was actually thinking about this a little bit more this morning and found this hacky "solution"/workaround (whatever you may call that): https://github.com/neovim/neovim/issues/15706#issuecomment-922253522

**Reply by sisrfeng** [reply]
I can not read code without conceal now.
I use a lot of matchadd() to define conceal, which can override syntax highlighting.
But it is growing slower and slower
